src/main/ts/ast/ts/DtsCleaner.ts

```diff
@@ -0,0 +1,53 @@
+import * as ts from 'typescript';
+import { serialize } from './Serializer';
+
+export interface CleanerOptions {
+  keepComments?: boolean;
+  keepVariables?: string[];
+  removeNamespaces?: string[];
+}
+
+const isNamed = (node: { name: ts.PropertyName | ts.BindingName }, names: string[]) =>
+  node.name.kind === ts.SyntaxKind.Identifier ? names.indexOf(node.name.text) !== -1 : false;
+
+const isVariableNamed = (statement: ts.VariableStatement, names: string[]) =>
+  statement.declarationList.declarations.some((decl) => isNamed(decl, names));
+
+const isExported = (node: { modifiers?: ts.ModifiersArray }): boolean =>
+  node.modifiers && node.modifiers.some((mod) => mod.kind === ts.SyntaxKind.ExportKeyword);
+
+// Strip out all comments, namespace, function and const declarations
+export const clean = (ast: ts.SourceFile, options: CleanerOptions = {}) => {
+  const keepVariables = options.keepVariables || [];
+  const removeNamespaces = options.removeNamespaces || [];
+
+  const nodes: ts.Node[] = [];
+  ts.forEachChild(ast, (node) => {
+    switch (node.kind) {
+      case ts.SyntaxKind.FunctionDeclaration:
+      case ts.SyntaxKind.MethodDeclaration:
+      case ts.SyntaxKind.PropertyDeclaration:
+        const decl = node as any;
```

Comment:
Is there a way to take this out before we merge it?

Reply:
We could, but it'd involve a type guard and removing the switch. This same logic is already used elsewhere, so I don't see the need and it's just caused by the AST design that doesn't play well with typescript (since checking the kind property doesn't act as a guard).